airflow/models/dagbag.py

```diff
@@ -466,6 +467,34 @@ def _bag_dag(self, *, dag, root_dag, recursive):
                         del self.dags[subdag.dag_id]
             raise
 
+    @provide_session
+    def _check_if_duplicate(self, dag, session=None):
+        """
+        Checks if a DAG with the same ID already exists.
+        If present, returns the fileloc of the existing DAG.
+        """
+        from airflow.models.serialized_dag import SerializedDagModel  # Avoid circular import
+
+        other_dag = session.query(SerializedDagModel).filter(
+            SerializedDagModel.dag_id == dag.dag_id,
+            SerializedDagModel.fileloc_hash != DagCode.dag_fileloc_hash(dag.fileloc)
+        ).first()
```

Comment:
In the unlikely event that there are more than two instances of a single DAG name, this is only going to alert on one of the collisions, correct?

Do you think its worthwhile to handle this case and alert for a larger collision, Or would it be too much added complexity for a rare failure?

Reply:
From the code, I believe that an import error will be raised for each collision. (If there are 3 DAGs with same ID, one gets parsed and the other 2 raise errors.)

Reply:
Yes, have tested and confirmed

Reply:
One case which I did miss was that of multiple DAGs with same ID in the same file. Have fixed that now.